rtp/src/main/kotlin/org/jitsi/rtp/rtp/RtpPacket.kt

```diff
@@ -273,9 +291,26 @@ open class RtpPacket(
         return newHeader
     }
 
+    private fun pickParserForEncodedHeaders(
+        pendingHeaderExtensions: List<PendingHeaderExtension>
+    ): HeaderExtensionParser? {
+        if (pendingHeaderExtensions.isEmpty()) {
+            return null
+        } else {
+            pendingHeaderExtensions.forEach {
+                if (it.id >= 15 || it.dataLengthBytes == 0 || it.dataLengthBytes > 16) {
+                    return TwoByteHeaderExtensionParser
+                }
+            }
+            return OneByteHeaderExtensionParser
+        }
+    }
+
     fun encodeHeaderExtensions() {
         val pendingHeaderExtensions = this.pendingHeaderExtensions ?: return
 
+        val newParser = pickParserForEncodedHeaders(pendingHeaderExtensions)
```

Comment:
If we have no header extensions at all we need to encode this (i.e. if we've dropped all header extensions from an existing packet).  Maybe I should have `pickParserForEncodedHeaders` return the one-byte header extension parser in this case, rather than `null`, and reserve null for unknown header extension type?

Reply:
Actually unknown type shouldn't happen here (for an unknown header type we wouldn't have been able to construct the pending header extensions in the first place) so just always returning 1-byte if we don't need 2-byte is fine.